.evergreen/handle-paths.sh

```diff
@@ -36,3 +36,7 @@ if [ -z "${DRIVERS_TOOLS:-}" ]; then
     DRIVERS_TOOLS=$(dirname $DRIVERS_TOOLS)
   done
 fi
+
+if [ "Windows_NT" = "${OS:-}" ]; then # Magic variable in cygwin
```

Comment:
`$OS` is a highly overloaded variable used by various scripts both in DET and by Drivers' own scripts. Recommend using `$OSTYPE` instead (Bash variable, less likely to be overwritten by user scripts):

```bash
if [[ "${OSTYPE:?}" = cygwin ]]; then
```

Reply:
Replaced all usages

Reply:
I ended up using `if [[ "$(uname -s)" == CYGWIN* ]]; then` consistently since it works from sh as well.

Reply:
I changing to `bash` would need to be a separate DRIVERS ticket, outside the scope of this change.

Reply:
Let's assume for now that only `run-orchestration.sh` and `download-mongodb.sh` need to be able to be run from `sh`, and inline `handle-paths.sh` into `run-orchestration.sh`.  I'll open up a ticket to ask drivers to audit their use of `sh` for drivers-tools scripts.

Reply:
On second thought, `handle-paths.sh` was too much to inline.  I opened https://jira.mongodb.org/browse/DRIVERS-2852.

Reply:
Note, `sh` _may_ invoke a `bash` script via `bash path/to/script.sh` so long as it does not require any variables to be modified/exported/etc. by `script.sh`. Similarly, `sh` may be invoked from a `bash` script via `sh path/to/script.sh` with similar restrictions. Since `handle-paths.sh` is designed to modify existing variables and is being sourced from so many scripts, I support changing the shebang to `sh` to indicate sh-compatibility.